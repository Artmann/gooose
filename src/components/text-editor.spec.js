import React from 'react';
import TextEditor from './text-editor';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe('TextEditor', () => {
  it('renders correctly in text mode', () => {
    const tree = renderer
      .create(<TextEditor text="**Hello World**"/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('updates the text', () => {
    const onChange = jest.fn();
    const textEditor = mount(<TextEditor text="foobar" onChange={ onChange } />);

    textEditor.find('[data-test-id="text-area"]').simulate('change', {
      target: {
        value: '**Hello World**'
      }
    });

    expect(onChange).toHaveBeenCalledWith('**Hello World**');
  });

  it('toggles between edit and preview mode', () => {
    const textEditor = mount(<TextEditor text="**Hello World**" />);

    expect(textEditor.find('[data-test-id="text-area"]')).toHaveLength(1);
    expect(textEditor.find('[data-test-id="preview"]')).toHaveLength(0);

    textEditor.find('[data-test-id="preview-button"]').simulate('click');

    expect(textEditor.find('[data-test-id="text-area"]')).toHaveLength(0);
    expect(textEditor.find('[data-test-id="preview"]')).toHaveLength(1);

    textEditor.find('[data-test-id="edit-button"]').simulate('click');

    expect(textEditor.find('[data-test-id="text-area"]')).toHaveLength(1);
    expect(textEditor.find('[data-test-id="preview"]')).toHaveLength(0);
  });

  it('displays the text', () => {
    const textEditor = mount(<TextEditor text="**Hello World**" />);

    expect(textEditor.find('[data-test-id="text-area"]').text()).toEqual('**Hello World**');
  });

  it('displays the text in preview mode', () => {
    const textEditor = mount(<TextEditor text="**Hello World**" />);

    textEditor.find('[data-test-id="preview-button"]').simulate('click');

    expect(textEditor.find('[data-test-id="preview"]').text().trim()).toEqual('Hello World');
  });
});
