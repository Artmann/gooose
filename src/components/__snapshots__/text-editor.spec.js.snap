// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor renders correctly in text mode 1`] = `
<div>
  <textarea
    className="sc-bdVaJa kxbDjW"
    data-test-id="text-area"
    onChange={[Function]}
    value="**Hello World**"
  />
  <div
    className="sc-bxivhb ElINa"
  >
    <button
      className="sc-htpNat kkVLjQ"
      data-test-id="edit-button"
      onClick={[Function]}
    >
      Write
    </button>
    <button
      className="sc-htpNat eKeXim"
      data-test-id="preview-button"
      onClick={[Function]}
    >
      Preview
    </button>
  </div>
</div>
`;
